include/swift/Basic/BasicBridging.h

```diff
@@ -13,6 +13,10 @@
 #ifndef SWIFT_BASIC_BASICBRIDGING_H
 #define SWIFT_BASIC_BASICBRIDGING_H
 
+// Workaround to avoid a compiler error because `cas::ObjectRef` is not defined
+// when including VirtualFileSystem.h
+#include "llvm/CAS/CASReference.h"
+
```

Comment:
What's requiring the actual definition? All the references I see are in `Optional`, which I thought would be fine with just a forward declaration. Is that not the case?

Reply:
This is due to some issue with Swift/C++ interop, see rdar://98579497

Reply:
We should cherry-pick `f93fb9e6953e12cd8f5eda6be9cd3dcd61de261f` in `stable/20221013`, but I'm still hesitant to put it in `next`, I think this is a bug with C++ interop that needs to be fixed.

Reply:
https://github.com/apple/llvm-project/pull/5102/commits/f93fb9e6953e12cd8f5eda6be9cd3dcd61de261f

Reply:
On second thought maybe having the change in Swift is appropriate since it's the Swift build that is affected.
But we should at least add a comment to explain that this is due to C++ interop and include the radar link.